controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java

```diff
@@ -55,35 +64,58 @@
 
     /** Create a curator db which also set up a ZooKeeper server (such that this instance is both client and server) */
     @Inject
-    public CuratorDb() {
-        // this.zooKeeperServer = new ZooKeeperServer(createZookeeperServerConfig());
-        // this.curator = new Curator("localhost:2281");
-        //this.zooKeeperServer = null;
-        this.curator = new MockCurator();
-    }
-    
-    private static ZookeeperServerConfig createZookeeperServerConfig() {
+    public CuratorDb(ClusterInfoConfig clusterInfo) {
+        this.zooKeeperServer = new ZooKeeperServer(toZookeeperServerConfig(clusterInfo));
+        this.curator = new Curator(toConnectionSpec(clusterInfo));
+    }
+
+    /** Create a curator db which does not set up a server, using the given Curator instance */
+    protected CuratorDb(Curator curator) {
+        this.zooKeeperServer = null;
+        this.curator = curator;
+    }
+
+    private static ZookeeperServerConfig toZookeeperServerConfig(ClusterInfoConfig clusterInfo) {
         ZookeeperServerConfig.Builder b = new ZookeeperServerConfig.Builder();
         b.zooKeeperConfigFile("conf/zookeeper/controller-zookeeper.cfg");
         b.dataDir("var/controller-zookeeper");
-        b.clientPort(2281);
+        b.clientPort(zooKeeperPort);
         b.myidFile("var/controller-zookeeper/myid");
-        b.myid(0);
-        ZookeeperServerConfig.Server.Builder server = new ZookeeperServerConfig.Server.Builder();
-        server.id(0);
-        server.hostname("localhost");
-        server.quorumPort(2282);
-        server.electionPort(2283);
-        b.server(server);
+        // Ideally we would inject and use the index in ClusterMemberInfoConfig, but the config is always empty for
+        // reasons unknown
```

Comment:
I suggest we replace with "... always empty since the config id used here is not the config id of the container, which is the config producer for this config"

Reply:
Actually not empty but null, which seems to be a bug to me, but I suggest we just remove both the ClusterMemberInfoConfig and the comment ...